### PR TITLE
refactor: Apply webpack-defaults

### LINKED ISSUE #477: refactor: apply webpack-defaults
Addition of `webpack-defaults` & associated refactoring as a part of the next `Major` release

Issue exists for status tracking across the organization. 

>Please do not close

### PR DESCRIPTION
**Pull request by d3viant0ne**
- [x] Apply `webpack-defaults` tooling updates
- [x] Apply `webpack-defaults` grammar updates
- [x] Apply `webpack-defaults` style updates
- [x] Convert existing tests to Jest from Mocha

The desired effect here is to `update` style & grammar without changing the way etwp works.

Note, this is being merged into #540 making this refactoring `etwp & webpack v3 only`

Closes #477

### CODE REVIEW

File: src/index.js [hunk 1]
```diff
@@ -0,0 +1,259 @@
...
+    if (isString(loader)) {
+      loader = loader.split('!');
+    }
+    if (isString(before)) {
+      before = before.split('!');
+    } else if (!Array.isArray(before)) {
+      before = [before];
+    }
+    options = mergeOptions({ omit: before.length, remove: true }, options);
+    delete options.loader;
```
Comment (michael-ciniawsky) [inline src/index.js]: I think we can remove it, since it was already deprecated in `v2.0.0` (*)

File: src/index.js [hunk 2]
```diff
@@ -0,0 +1,259 @@
...
+    delete options.use;
+    delete options.fallback;
+    delete options.fallbackLoader;
```
Comment (michael-ciniawsky) [inline src/index.js]: \*

File: src/index.js [hunk 3]
```diff
@@ -0,0 +1,259 @@
...
+    }
+    if (options.loader) {
+      console.warn('loader option has been deprecated - replace with "use"');
+    }
+    if (Array.isArray(options) || isString(options) || typeof options.options === 'object' || typeof options.query === 'object') {
+      options = { loader: options };
+    } else {
+      validateOptions(path.resolve(__dirname, './schema/loader.json'), options, 'Extract Text Plugin (Loader)');
+    }
+    let loader = options.use || options.loader;
```
Comment (michael-ciniawsky) [inline src/index.js]: \*

```diff
+ options.use
- options.use || options.loader
```

File: src/index.js [hunk 4]
```diff
@@ -0,0 +1,259 @@
...
+    let before = options.fallback || options.fallbackLoader || [];
```
Comment (michael-ciniawsky) [inline src/index.js]: \*

```diff
+ options.fallback || []
- options.fallback || options.fallbackLoader || []
```

File: src/index.js [hunk 5]
```diff
@@ -0,0 +1,259 @@
...
+        'Example: if your old code looked like this:\n' +
+        '    ExtractTextPlugin.extract("style-loader", "css-loader")\n\n' +
+        'You would change it to:\n' +
+        '    ExtractTextPlugin.extract({ fallback: "style-loader", use: "css-loader" })\n\n' +
+        'The available options are:\n' +
+        '    use: string | object | loader[]\n' +
+        '    fallback: string | object | loader[]\n' +
+        '    publicPath: string\n');
+    }
+    if (options.fallbackLoader) {
```
Comment (michael-ciniawsky) [inline src/index.js]: \*

```diff
-   if (options.fallbackLoader) {
-      console.warn('fallbackLoader option has been deprecated - replace with "fallback"');
-   }
```

File: src/index.js [hunk 6]
```diff
@@ -0,0 +1,259 @@
...
+      console.warn('fallbackLoader option has been deprecated - replace with "fallback"');
+    }
+    if (options.loader) {
```
Comment (michael-ciniawsky) [inline src/index.js]: \*

```diff
-   if (options.loader) {
-      console.warn('loader option has been deprecated - replace with "use"');
-   }
```

File: test/webpack-integration.test.js [hunk 7]
```diff
@@ -1,13 +1,13 @@
...
+import vm from "vm";
+import path from "path";
+import webpack from "webpack";
+import ExtractTextPlugin from '../src';
 
 var cases = process.env.CASES ? process.env.CASES.split(",") : fs.readdirSync(path.join(__dirname, "cases"));
 
-describe("TestCases", function() {
+describe("Webpack Integration Tests", function() {
 	cases.forEach(function(testCase) {
```
Comment (sokra) [inline test/webpack-integration.test.js]: spacing in this file is inconsistent

**Comment by d3viant0ne** [comment]
Thanks Tobias, I'll get everything updated this evening.

**Pull request merged**